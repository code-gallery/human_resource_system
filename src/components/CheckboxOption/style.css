.CheckboxOption {
  box-sizing: border-box;
  position: relative;
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
}

.CheckboxOption__input {
  margin-right: 5px;
}

.CheckboxOption__text {
  margin: 0;
  font-family: 'Graphik-Regular', sans-serif;
  -webkit-font-smoothing: antialiased;
  /* font-size: 10px; */
  line-height: 14px;
  margin-right: 10px;
  color: #7F8DAA;
}
