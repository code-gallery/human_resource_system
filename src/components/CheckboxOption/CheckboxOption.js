import React from 'react'
import PropTypes from 'prop-types'
import Checkbox from 'components/Checkbox'
import './style.css'

const CheckboxOption = ({ checked, text, onChange, className }) => (
  <div className={`CheckboxOption ${className}`}>
    <Checkbox
      className="CheckboxOption__input"
      checked={checked}
      onChange={onChange}
    />

    <p className="CheckboxOption__text">{text}</p>

  </div>

)

CheckboxOption.propTypes = {
  text: PropTypes.string.isRequired,
  checked: PropTypes.bool,
  onChange: PropTypes.func.isRequired,
  className: PropTypes.string
}

CheckboxOption.defaultProps = {
  className: '',
  checked: false
}

export default CheckboxOption
