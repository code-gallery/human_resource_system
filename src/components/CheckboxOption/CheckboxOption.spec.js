import React from 'react'
import renderer from 'react-test-renderer'
import CheckboxOption from './CheckboxOption'

describe('<CheckboxOption />', () => {
  const props = {
    text: 'Do want to win?',
    onChange: jest.fn()
  }

  describe('Rendering', () => {
    it('renders', () => {
      const tree = renderer.create(<CheckboxOption {...props} />).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
