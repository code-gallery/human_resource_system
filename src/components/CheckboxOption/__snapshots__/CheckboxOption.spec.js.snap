// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckboxOption /> Rendering renders 1`] = `
<div
  className="CheckboxOption "
>
  <div
    className="Checkbox CheckboxOption__input"
  >
    <label
      className="Checkbox__label"
    >
      <input
        checked={false}
        className="Checkbox__input"
        onChange={[Function]}
        type="checkbox"
      />
    </label>
  </div>
  <p
    className="CheckboxOption__text"
  >
    Do want to win?
  </p>
</div>
`;
